### PR TITLE
PT-311 Hydrophone service return HTTP 200 when the SES email service returned an error

### PR DESCRIPTION
**Pull request by yvon-dblg**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,15 +1,19 @@
 # Hydrophone
 
-Hydrophone is the module responsible of sending emails. 
-This API sends notifications to users for things like forgotten passwords, initial signup, and invitations. 
+Hydrophone is the module responsible of sending emails.
```
Comment (BigFab) [inline CHANGELOG.md]:
```suggestion
Hydrophone is the module responsible for sending emails.
```

File: api/forgot.go [hunk 2]
```diff
@@ -107,7 +107,8 @@ func (a *Api) passwordReset(res http.ResponseWriter, req *http.Request, vars map
 		} else {
 			a.logMetricAsServer("reset confirmation failed to be sent")
 			log.Print("Something happened generating a passwordReset email")
-			// TODO: do not respond with http.StatusOK (see Jira PT-311)
+			res.WriteHeader(http.StatusUnprocessableEntity)
```
Comment (BigFab) [inline api/forgot.go:110]: is this really a client error? or a server error? should it not be a 5xx?

File: test.sh [hunk 3]
```diff
@@ -1,5 +1,8 @@
 #!/bin/sh -eu
 
+export TEMPLATE_PATH="$(dirname $(readlink -f $0))/templates"
+echo "TEMPLATE_PATH=${TEMPLATE_PATH}"
```
Comment (BigFab) [inline test.sh]:
```suggestion
```

**Review by BigFab — COMMENTED** [review]
I was able to test the error raising with /autoconf

Before the fix:

image

After the fix:

image

**Comment by BigFab** [comment]
I made some suggestions before finally approving the PR. Overall it looks good to me. Very good job ! 👍

File: api/forgot.go (hunk 2 above)
Comment (yvon-dblg) [reply api/forgot.go:110]: As mentioned in the Jira issue:

> HTTP Error StatusUnprocessableEntity (422)
> The server understands the content type of the request entity (hence a 415 Unsupported Media Type status code is inappropriate), and the syntax of the request entity is correct (thus a 400 Bad Request status code is inappropriate) but was unable to process the contained instructions.

This description seems appropriate to me. Sometimes yes, a 500 may be better, but not always. This error can have multiple origin.

**Review by BigFab — APPROVED** [review]
OK for me

**Pull request merged**